app/controllers/rents_controller.rb

```diff
@@ -0,0 +1,13 @@
+class RentsController < ApplicationController
+  include Wor::Paginate
+  before_action :authenticate_user!, only: %i[index create]
+
+  def index
+    render_paginated Rent.all
+  end
+
+  def create
+    rent = Rent.create!(params.require(:rent).permit(:user_id, :book_id, :start_date, :end_date))
```

Comment:
Move this line to another method
```rb
def create
  rent = Rent.create!(create_rent_params)
  render json: rent
end

private 

def create_rent_params
  params.require(:rent).permit(:user_id, :book_id, :start_date, :end_date)
end
```

Reply:
I understand that we separate it so it's more comprehensible, but why do we have to put it in the private section?

Reply:
[Encapsulamiento](https://bandageek.com/2016/06/programacion-orientada-a-objetos-caracteristicas/)
![image](https://user-images.githubusercontent.com/11413458/55733170-be87b300-59f3-11e9-8c66-de0486b166dd.png)
Una de las técnicas de abstracción que brinda la programación orientada a objetos es el encapsulamiento. Esta permite restringir el acceso a ciertos atributos y métodos. La idea principal de este recurso es ocultar al desarrollador información que utiliza una clase para funcionar permitiéndole trabajar solo con las variables y funciones que este realmente necesita.

La idea es que los metodos que no queres que tu clase exponga los deberias declarar como privados. Sobre todo los metodos privados complementan (o son necesarios) para que la ejecucion de un metodo publico sea mas expresivo. Ejemplo:
```rb
class Demo
  def create_and_send_reports
    reports = create_reports
    send_by_email(reports)
  end

  private

  def create_reports
    # TODO: pending to implement
  end

  def send_by_email
    # TODO: pending to implement
  end
end
```
Queremos que la clase Demo exponga el metodo create_and_send_reports independientemente de lo que tenga que hacer adentro

Volviento a este ejemplo en particular: no queremos que el controller exponga el metodo create_rent_params